tools/gen_struct_info.py

```diff
@@ -266,6 +266,7 @@ def inspect_headers(headers, cflags):
                                '-Wno-format',
                                '-nostdlib',
                                compiler_rt,
+                               '-s', 'MEMORY64=' + str(settings.MEMORY64),
```

Comment:
I don't think this will work.      This gets run just once one first use so we don't want to depend on the settings use in that moment.    If there are differences here then we would need a second set of json file and have them stored separately in the cache.  

See `generate_struct_info` in `emscripten.py`.

Reply:
Ok, I guess I am not clear on how to fix this.. maybe we can discuss that offline. Or at least maybe I can just put a FIXME in there for when testing with `wasm64` resumes.. so far this has worked in allowing tests to pass.

Reply:
The fix I think would be to have a different filename when running in 64bit mode.  See:
https://github.com/emscripten-core/emscripten/blob/bca56b3956549f068f62590d643c01f5e067713d/emscripten.py#L826

It won't work as it stands because if you have previously built the struct_info.json with `-sMEMORY64` and then try to build something without that flag .. it would use the wasm64 version of the json file.. and things be broken in horrible ways, right?

Reply:
Ok, so we need a `generated_struct_info64.json` ? I can look into how to set that up in a follow-up. I'll comment this out for now.